akka-actor-tests/src/test/scala/akka/util/ByteStringSpec.scala

```diff
@@ -617,5 +618,10 @@ class ByteStringSpec extends WordSpec with Matchers with Checkers {
         }
       }
     }
+    "return the correct class from Builder.result" in {
+      val builder = ByteString().genericBuilder[String]
```

Comment:
Is there any reason why this method should not throw an UnsupportedOperationException?

Use ByteString.newBuilder instead?